### PR TITLE
MOTECH-1548: IVR: default configuration

### PR DESCRIPTION
**Pull request by matkwiatkowski**

### CODE REVIEW

**Comment by matkwiatkowski** [comment]
test this please

**Comment by matkwiatkowski** [comment]
test this please

File: ivr/src/main/java/org/motechproject/ivr/service/OutboundCallService.java [hunk 1]
```diff
@@ -14,4 +14,11 @@
      * @param params the parameters that will be used for building the url, the ones that don't match the placeholders will be sent as params of the request
      */
     void initiateCall(String configName, Map<String, String> params);
+
+    /**
+     * Gets the defaultConfig and calls the initiateCall method.
+     * Throws ConfigNotFoundException when there is no default configuration.
```
Comment (tstalka) [inline ivr/src/main/java/org/motechproject/ivr/service/OutboundCallService.java]: use @throws, and place it below @param

File: ivr/src/main/java/org/motechproject/ivr/service/impl/ConfigServiceImpl.java [hunk 2]
```diff
@@ -36,26 +35,28 @@
     private static final String CONFIG_FILE_PATH = "/org.motechproject.ivr/raw/" + CONFIG_FILE_NAME;
     private static final Logger LOGGER = LoggerFactory.getLogger(ConfigServiceImpl.class);
     private SettingsFacade settingsFacade;
-    private Map<String, Config> configs = new HashMap<>();
+    private Configs configs;
+    private Map<String, Config> configMap = new HashMap<>();
+    private String defaultConfig;
```
Comment (tstalka) [inline ivr/src/main/java/org/motechproject/ivr/service/impl/ConfigServiceImpl.java]: We can use configs object to get defaultConfig name

File: ivr/src/main/resources/webapp/js/controllers.js [hunk 3]
```diff
@@ -187,9 +204,16 @@
...
             .success(function(response){
+                var i;
                 $scope.configs = response;
                 $scope.originalConfigs = angular.copy($scope.configs);
-                setAccordions($scope.configs);
+                setAccordions($scope.configs.configList);
+                for (i = 0 ; i < response.configList.length ; i = i + 1) {
+                   if (response.defaultConfig === response.configList[i].name) {
+                      $scope.defaultConfigIndex = i;
+                         break;
```
Comment (tstalka) [inline ivr/src/main/resources/webapp/js/controllers.js]: bad indentation

File: ivr/src/main/resources/webapp/messages/messages.properties [hunk 4]
```diff
@@ -83,6 +83,7 @@ ivr.web.settings.delete=Delete
 ivr.web.settings.cancel=Cancel
 ivr.web.settings.save=Save
 ivr.web.settings.saved=Settings updated successfully
+ivr.web.settings.set_default=Set Default
```
Comment (tstalka) [inline ivr/src/main/resources/webapp/messages/messages.properties]: Set as default

File: ivr/src/main/resources/webapp/js/controllers.js [hunk 5]
```diff
@@ -187,9 +204,16 @@
 
         $http.get('../ivr/ivr-configs')
             .success(function(response){
+                var i;
                 $scope.configs = response;
                 $scope.originalConfigs = angular.copy($scope.configs);
-                setAccordions($scope.configs);
+                setAccordions($scope.configs.configList);
```
Comment (tstalka) [inline ivr/src/main/resources/webapp/js/controllers.js]: parameter will be undefinded?

File: ivr/src/main/resources/webapp/js/controllers.js [hunk 6]
```diff
@@ -203,10 +227,20 @@
             autoExpandSingleAccordion();
         };
 
+        $scope.setDefaultConfig = function(name, index) {
+            $scope.configs.defaultConfig = name;
+            $scope.defaultConfigIndex = index;
+        };
+
+        $scope.keepDefaultConfig = function() {
```
Comment (tstalka) [inline ivr/src/main/resources/webapp/js/controllers.js:235]: unused

Comment (matkwiatkowski) [inline ivr/src/main/resources/webapp/js/controllers.js:235]: actually it is used in settings.html to set current config as default

**Comment by tstalka** [comment]
test this please

**Pull request merged**